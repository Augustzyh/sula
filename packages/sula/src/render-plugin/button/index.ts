import ButtonPlugin, { LoadingButton } from './Button';
import { ButtonProps } from './Button';

export type Props = ButtonProps;
export default LoadingButton;
export { ButtonPlugin };
