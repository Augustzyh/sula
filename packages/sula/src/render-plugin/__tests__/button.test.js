import React from 'react';
import { mount } from 'enzyme';
import { ButtonPlugin } from '../button';
import '../../__tests__/common';

describe('button', () => {
  jest.useFakeTimers(); // 设置快速时间
  it('basic', () => {
    const actions = jest.fn();
    const wrapper = mount(<ButtonPlugin
      icon="tablet"
      config={{
        action: [
          actions,
          'stringAction'
        ]
      }}
    >test</ButtonPlugin>)
    expect(wrapper.render()).toMatchSnapshot();
  })
  
  it('button click, no lastAction.final', () => {
    const actions = jest.fn();
    const wrapper = mount(
      <ButtonPlugin
        autoLoading={false}
        config={{
          action: [
            ctx => {
              ctx.button.showLoading();
              setTimeout(() => {
                ctx.button.hideLoading()
              }, 2000);
            },
            actions,
            'finaAction'
          ]
        }}
      >按钮</ButtonPlugin>
    )
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.find(ButtonPlugin).at(0).simulate('click');
    jest.runAllTimers(); // 时间快进
    expect(wrapper.find('button').text()).toBe('按 钮');
    expect(actions).toBeCalled();
  })

  it('button click, has lastAction.final', () => {
    const actions = jest.fn();
    const wrapper = mount(
      <ButtonPlugin
        config={{
          action: [
            actions,
            {
              type: jest.fn(),
              final: jest.fn()
            }
          ]
        }}
      >按钮</ButtonPlugin>
    )
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.find(ButtonPlugin).at(0).simulate('click');
    jest.runAllTimers(); // 时间快进

    expect(wrapper.find('button').text()).toBe('按 钮');
    expect(actions).toBeCalled();

    wrapper.update();
    wrapper.unmount();
  })

  it('button click, has lastAction.final', () => {
    const actions = jest.fn();
    const wrapper = mount(
      <ButtonPlugin
        config={{
          action: [
            actions,
            {
              type: jest.fn(),
              before: jest.fn()
            }
          ]
        }}
      >按钮</ButtonPlugin>
    )
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.find(ButtonPlugin).at(0).simulate('click');
    jest.runAllTimers(); // 时间快进

    expect(actions).toBeCalled();
  })
})
