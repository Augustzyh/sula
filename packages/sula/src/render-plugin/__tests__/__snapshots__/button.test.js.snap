// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button basic 1`] = `
<button
  class="ant-btn"
  type="button"
>
  <span>
    test
  </span>
</button>
`;

exports[`button button click, has lastAction.final 1`] = `
<button
  class="ant-btn"
  type="button"
>
  <span>
    按 钮
  </span>
</button>
`;

exports[`button button click, has lastAction.final 2`] = `
<button
  class="ant-btn"
  type="button"
>
  <span>
    按 钮
  </span>
</button>
`;

exports[`button button click, no lastAction.final 1`] = `
<button
  class="ant-btn"
  type="button"
>
  <span>
    按 钮
  </span>
</button>
`;
